client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/ups-settings-form.js

```diff
@@ -191,12 +191,12 @@ const UpsSettingsForm = ({ translate, errorNotice, successNotice, countryNames,
 					type: 'UpsAccount',
 				};
 
-				if( isInvoiceDetailsChecked ) {
```

Comment:
The form values were being deleted upon submission if the checkbox was checked.
We should _not_ send the values upon submission if the checkbox is _not_ checked.

Reply:
It seems strange we did it this way. Do you know why we delete when it's check?

Reply:
When the checkbox is checked and then unchecked, we don't clear the values for the invoice data in the `formValues` object.
So if the merchant checks the checkbox, then enters some values for the invoice details, then unchecks the checkbox, the "hidden" invoice values would otherwise be sent in the request.
But we also want to "keep" the values previously entered in mind if the merchant unchecks the checkbox by mistake.

Let me know if it makes sense, it's not too straightforward :D

Reply:
Ah I see, thanks. In this branch, if I enter check the box, enter invoice data, uncheck the box by mistake, check it again, all the invoice data is gone. This is still okay right?

Reply:
Thanks for pointing it out - that's weird, it must have been missed. I fixed it in 50175438. I tried adding a test, but enzyme doesn't do a good job at representing the values of the field. RTL should be better, tho!